tests/test_routes/test_environment_route.py

```
@@ -0,0 +1,244 @@
+        assert response.status_code == 404
+        assert response.json() == {"detail": "Environment not found"}
+        mock_update_env.assert_called_once_with(
+            "123", Env(latitude=1, longitude=1)
+        )
+
+def test_update_env_server_error():
+    with patch.object(
+        EnvController,
+        "update_env_by_id",
+        side_effect=Exception("error"),
+    ) as mock_update_env:
```

Ruff

Local variable `mock_update_env` is assigned to but never used

(F841)